File: site/swingjs/j2s/main/java/Ionex/AmountTextField.js
(function(){var P$=Clazz.newPackage("main.java.Ionex");
/*c*/var C$=Clazz.newClass(P$, "AmountTextField", null, 'javax.swing.JTextField');

C$.$clinit$=2;

Clazz.newMeth(C$, '$init$', function () {
},1);

C$.$fields$=[['S',['lastValue']]]

Clazz.newMeth(C$, 'c$$S',  function (string) {
;C$.superclazz.c$$S.apply(this,[string]);C$.$init$.apply(this);
}, 1);

Clazz.newMeth(C$, 'keyUp$java_awt_Event$I',  function (evt, key) {
var strText=this.getText$().trim$();
if (strText.equals$O("")) {
this.lastValue=this.getText$();
return true;
}try {
var intValue=Integer.valueOf$S(this.getText$().trim$()).intValue$();
if ((intValue > 0) && (intValue <= 400) ) {
this.lastValue=this.getText$();
} else {
this.setText$S(this.lastValue);
}} catch (e) {
if (Clazz.exceptionOf(e,"NumberFormatException")){
this.setText$S(this.lastValue);
} else {
throw e;
}
}
return true;
});

Clazz.newMeth(C$);
})();
;Clazz.setTVer('3.3.1-v4');//Created 2022-04-21 16:53:54 Java2ScriptVisitor version 3.3.1-v4 net.sf.j2s.core.jar version 3.3.1-v4

File: src/main/java/Ionex/AmountTextField.java
package main.java.Ionex;

import javax.swing.JTextField;
import java.awt.Event;

class AmountTextField extends JTextField {
    /**
	 * 
	 */
	private static final long serialVersionUID = 3905619255423613492L;
	String lastValue;

    public AmountTextField(String string) {
        super(string);
    }

    public boolean keyUp(Event evt, int key) {
        //if the field is now empty, it's OK
        String strText = getText().trim();
        if (strText.equals("")) {
            lastValue = getText();
            return true;
        }

        //see if the value is a valid integer
        try {
            int intValue = Integer.valueOf(getText().trim()).intValue();
            if ((intValue > 0) && (intValue <= 400)) {
                lastValue = getText();
            } else {
                setText(lastValue);
//					Toolkit.getDefaultToolkit().beep();
            }
        } catch (NumberFormatException e) {
            setText(lastValue);
//				Toolkit.getDefaultToolkit().beep();
        }

        return true;
    }
}